src/node_snapshotable.cc

```diff
@@ -956,6 +956,16 @@ const SnapshotData* SnapshotBuilder::GetEmbeddedSnapshotData() {
 )";
 }
 
+// Reset context settings that need to be initialized again after
+// deserialization.
+static void ResetContextSettingsBeforeSnapshot(Local<Context> context) {
+  // Reset the AllowCodeGenerationFromStrings flag to true (default value) so
+  // that it can be re-initialized with v8 flag
+  // --disallow-code-generation-from-strings and recognized in
+  // node::InitializeContextRuntime.
+  context->AllowCodeGenerationFromStrings(true);
```

Comment:
Do we need this? Doesn't V8 always reset this value during deserialization, and we just need to propagate that value to the embedder data?

Reply:
IIUC, this value is saved in the [native context field](https://source.chromium.org/chromium/chromium/src/+/main:v8/src/objects/contexts.h;l=86?q=ALLOW_CODE_GEN_FROM_STRINGS_INDEX) so it is saved in the snapshot. After the deserialization, V8 only set the value to false if flag `--disallow-code-generation-from-strings ` is [present](https://source.chromium.org/chromium/chromium/src/+/main:v8/src/init/bootstrapper.cc;l=6511?q=allow_code_gen_from_strings). So to restore the default value `true` (i.e. allow codegen from strings), we have to reset the value here.

Reply:
Oh I see, V8 only resets that from true to false with the flag (or false to false), it doesn't reset it from false to true even if you use `--no-disallow-code-generation-from-strings`.

I suppose we can also just save the value of `--disallow-code-generation-from-strings` when we parse the arguments, just like what we do for `--no-harmony-import-assertions` and `--abort_on_uncaught_exception` etc. in `ProcessGlobalArgs()` and use that in `InitializeContextRuntime()` instead of relying on `context-> IsCodeGenerationFromStringsAllowed()`. Another alternative is to add an else branch in V8 to set that field to true when the flag is false. But I am also okay with the current approach, so I'll defer to you to decide - if the current approach is chosen please move the reset calls to where the initialization ends (preferably in the context scope if there is one), to make the code look less scattered.

Reply:
Thanks for the suggestion. Updated.